### ISSUE TITLE
Class is not added in the component when the style is inside the template

### ISSUE THREAD

**Issue by danicomas** [angular/angular#3868]
```js
@Component({
  selector: 'ngdialog',
  host: {
    class: 'dialog-container' //is not added in the DOM
  }
})
@View({
  templateUrl: 'dialog.html' //the class host is here. Is it wrong?
})

// Component controller
class Dialog {
  title:string;
  constructor() {
    this.title = "dialog"
  } 
}
```

```html
<style>
  .dialog-content {
    border: 1px solid black;
    background: gray;
    height: 100px;
    width: 200px;
    margin: 0 auto;
  }
  
  .dialog-container {
    position: absolute;
    width: 100%;
    text-align: center;
    top: 5%;     
  }
  
</style>

<div class="dialog-content"> <!--the class dialog-content is binded correctly-->
  {{title}}
</div>
```

**Comment by danicomas** [comment]
I have tried this and doesn't work:

```js
@View({
  templateUrl: 'dialog.html',
  styleUrls: ['dialog.css']
})
```

The class is not binding in the primary element :S

**Comment by danicomas** [comment]
It works with the following:

```js
@Component({
  selector: 'ng-dialog',
  host: {
    class: 'dialog-container'
  }
})
@View({
  templateUrl: 'dialog.html',
  styleUrls: ['dialog.css'],
  encapsulation: ViewEncapsulation.NONE
})

// Component controller
class Dialog {
  title:string;
  constructor() {
    this.title = "dialog"
  } 
}
```

**Comment by pkozlowski-opensource** [comment]
@danicomas it is a bit hard to follow what your problem is / what you are struggling with. Are you saying that you think you've found a bug? Or you've got general support questions / need clarification? In the later case, could you please try to use gitter channel first (https://gitter.im/angular/angular) for support requests?

If you believe that you are bumping into a bug, could you please share a minimal reproduce scenario using plunker or similar? Thnx!

**Comment by danicomas** [comment]
@pkozlowski-opensource I don't know if it is a bug or no. The first comment and the second is about the class of the first element wasn't binded but the child class of the child elements yes.

```html
<ng-dialog class="ng-binding dialog-container" _nghost-dqg-0=""> <!-- first element-->
<div class="dialog-content ng-binding" _ngcontent-dqg-0=""> <!-- child-->
  kk
</div>
</ng-dialog>
```

 Then, the last comment is about I have found a solution adding encapsulation attribute to NONE

```html
<ng-dialog class="ng-binding dialog-container"> <!-- first element-->
<div class="dialog-content ng-binding"> <!-- child-->
  kk
</div>
</ng-dialog>
```

but I don't know if it is correct. ViewEncapsulation.NONE causes that the elements are not binded but is it the solution or should it be binded without encapsulation?

**Comment by cghislai** [comment]
I think if you put your CSS class on the host attribute of the ng-dialog component, then it will be added to the <ng-dialog> tag. DOM shadowing will hide this from component's view. If you place your .dialog-ocntainer CSS rules in the parent component of ng-dialog, then it should work.

To keep your style in your component scope, why not just wrap your template in a <div class=dialog-container>?

**Comment by danicomas** [comment]
@cghislai I have put it in the parent element and it doesn't work. Probably the binding is about _ngcontent-<componentid> and the componentid is not the same :S

In angular 2 we don't have a replace attribute. If I wrap this: ```<div class=dialog-container>``` the result will be:

```html
<ngdialog>
<div class=dialog-container>
</div>
</ngdialog
```

For the modal I need the position:absolute :D

By the moment I think that the best way is this solution: https://github.com/angular/angular/issues/3868#issuecomment-135697559

**Comment by mhevery** [comment]
Sounds like this has been resloved. Closing.
